matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPhenotypesMap.java

```diff
@@ -219,17 +219,29 @@ public static void reorder(List<Map<String, String>> predefined1, List<Map<Strin
             if (StringUtils.equals(name1, name2)) {
                 common1.add(item1);
```

Comment:
I understand...  I wonder if removing `next1` and `next2` and combining the `if` and `while` after the loop may decrease the complexity enough to avoid excllusions

(e.g. instead of separate if and while have something like
```while (item1 != null) {
              unique1.add(item1);
              item1 = iterator1.hasNext()? iterator1.next() : null;
    }
```